### PR TITLE
Feat/82 reconnection UI

### PR DESCRIPTION
**Pull request by simbayippy**
this PR builds on top of the previous PR #100 as it requires the proper linting types

Goal:
- connection UI toast when other party joins
- disconnection UI toast when the other party leaves
- allow for reconnection logic

https://github.com/user-attachments/assets/0363943c-72f7-46bb-9dcc-cf0fc775e08b

### CODE REVIEW

File: peerprep-fe/src/app/collaboration/components/CollaborationEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  const bindingRef = useRef<MonacoBinding | null>(null);
+  const editorRef = useRef<MonacoEditor.IStandaloneCodeEditor | null>(null);
+  const prevClientsRef = useRef<Map<number, ConnectedClient>>(new Map());
+  const sockServerURI =
+    process.env.NEXT_PUBLIC_SOCK_SERVER_URL || 'ws://localhost:4444';
+  const { toast } = useToast();
+  const { clearLastMatchId } = useCollaborationStore();
+  const router = useRouter();
+
+  const stringToColor = (str: string) => {
```
Comment (Kurtyjlee) [inline peerprep-fe/src/app/collaboration/components/CollaborationEditor.tsx:63]: minor nit: you might want to shift this code into utils to allow for reusability

File: peerprep-fe/src/app/collaboration/page.tsx [hunk 2]
```diff
@@ -1,44 +1,52 @@
...
-import { editor as MonacoEditor } from 'monaco-editor';
-
-const CollaborationPage = () => {
-  const [selectionProblem, setSelectionProblem] = useState<Problem | null>(null);
+import dynamic from 'next/dynamic';
+import EditorSkeleton from './components/EditorSkeleton';
+import LoadingSpinner from '@/components/loading/LoadingSpinner';
+import { useCollaborationStore } from '@/state/useCollaborationStore';
+
+const CollaborationEditor = dynamic(
```
Comment (Kurtyjlee) [inline peerprep-fe/src/app/collaboration/page.tsx:21]: not sure that dynamic does for this case, do we need it to lazy load?

File: peerprep-fe/src/hooks/use-toast.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+const TOAST_REMOVE_DELAY = 1000000;
+
+type ToasterToast = ToastProps & {
+  id: string;
+  title?: React.ReactNode;
+  description?: React.ReactNode;
+  action?: ToastActionElement;
+};
+
+const actionTypes = {
```
Comment (Kurtyjlee) [inline peerprep-fe/src/hooks/use-toast.ts:18]: nit: might want to turn this into a type since it is only used as a type

**Review by Kurtyjlee — COMMENTED** [review]
Feature works well, found some minor nits and ui issues

File: peerprep-fe/src/app/collaboration/page.tsx (hunk 2 above)
Comment (simbayippy) [reply peerprep-fe/src/app/collaboration/page.tsx:21]: hmm reason was that if i didnt do this there were some build errors

Comment (Kurtyjlee) [reply peerprep-fe/src/app/collaboration/page.tsx:21]: :o HAHA thats so weird

**Review by Kurtyjlee — APPROVED** [review]
LGTM

**Review by Kurtyjlee — APPROVED** [review]
LGTM

**Pull request merged**